tests/fallback_test.py

```diff
@@ -133,7 +134,19 @@ def assert_ads_connections(
         primary_status: channelz_pb2.ChannelConnectivityState,
         fallback_status: channelz_pb2.ChannelConnectivityState,
     ):
-        self.assertEqual(
+        self.assertTrue(
+            self.ads_connections_status_check_result(
+                client, primary_status, fallback_status
+            )
+        )
+
+    def ads_connections_status_check_result(
+        self,
+        client: framework.helpers.docker.Client,
+        primary_status: channelz_pb2.ChannelConnectivityState,
+        fallback_status: channelz_pb2.ChannelConnectivityState,
```

Comment:
bad type hint

<img width="604" height="136" alt="Image" src="https://github.com/user-attachments/assets/90d29f1d-f62a-41e9-bfde-8a59ac5aa25c" />

```suggestion
        primary_status: channelz_pb2.ChannelConnectivityState.State,
        fallback_status: Optional[channelz_pb2.ChannelConnectivityState.State],
```

assert_ads_connections signature needs to be updated too

Reply:
Yea, I see you pretty much copied the original signature. We should add a type checker CI to avoid problems like this in the first place.

Reply:
Done.